File: datastructures lab/practice/QUEUE_LL.c
/******************************************************************************
    C Program To Implement QUEUE Using Linked List.
*******************************************************************************/

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *link;
};
struct node *first = NULL;

void insert()
{
    int item;
    struct node *p,*temp;
    p = malloc(sizeof(struct node));
    printf("Enter The Element To be Pushed : ");
    scanf("%d",&item);
    p->data = item;
    p->link = NULL;
    
    if(first == NULL)
        first = p;
    else
    {
        temp = first;
        while(temp->link != NULL)
        {
            temp = temp->link;
        }
        temp->link = p;
    }
}

void delet()
{
    struct node *temp;
    if(first == NULL)
        printf("POP Operation Not Possibl.!");
    else if(first->link == NULL)
    {
        printf("Element Deleted is %d",first->data);   
        free(first);
        first = NULL;
    }
    else
    {
        temp = first;
        first = first->link;
        printf("Element Deleted is %d",temp->data);
        free(temp);
    }
}

void display()
{
    struct node *temp;
    if(first == NULL)
        printf("Display Operation Not Possible.!");
    else
    {
        temp = first;
        while(temp != NULL)
        {
            printf("\t%d",temp->data);
            temp = temp->link;
        }
    }
}

int main()
{
    int choice;
    printf("QUEUE Implementation Using Linked List : ");
    while(1)
    {
        printf("\nThe Choices Are : \n1. Insert.\n2. Delete.\n3. Display.\n4. Exit.\nEnter Your Choice : ");
        scanf("%d",&choice);
        
        switch(choice)
        {
            case 1 : insert();
                     break;
                     
            case 2 : delet();
                     break;
                     
            case 3 : display();
                     break;
                     
            case 4 : exit(0);
                     break;
                     
            default : printf("Invalid choice.!");         
        }
    }
    return 0;
}

File: OOJava_lab/DataStructures/QUEUE_LL.java
/******************************************************************************
    Java Program To implement QUEUE Using Linked List.        
*******************************************************************************/

import java.util.Scanner;
import java.util.LinkedList;

class Queue
{
    LinkedList<Integer> Q = new LinkedList<Integer>();
    
    public void insert()
    {
        System.out.println("Enter The Item To Be Inserted : ");
        Scanner sc = new Scanner(System.in);
        int item = sc.nextInt();
        Q.addLast(item);
    }
    
    public void delet()
    {
        if(Q.size() == 0)
        {
            System.out.println("POP Operation Not Possible, QUEUE UNDERFLOW.");
        }
        else
        {
            System.out.println("Element Deleted is "+Q.removeFirst());
        }
    }
    
    public void display()
    {
        if(Q.size() == 0)
        {
            System.out.println("Display Operation Not Possible, QUEUE is EMPTY.!");
        }
        else
        {
            System.out.println(Q);   
        }
    }
}

public class QUEUE_LL
{
	public static void main(String[] args) 
	{
		Queue q = new Queue();
		
		System.out.println("QUEUE Impelmentation Using LinkedList : ");
		
		while(true)
		{
		    System.out.println("The Chocies Are : ");
		    System.out.println("1. INSERT. \n2. DELETE. \n3. Display. \n4. Exit");    
		    System.out.println("\nEnter Your Choice : ");
		    Scanner sc = new Scanner(System.in);
		    int choice = sc.nextInt();
		    
		    switch(choice)
		    {
		        case 1 : q.insert();
		                 break;
		               
		        case 2 : q.delet();
		                 break;
		                 
		        case 3 : q.display();
		                 break;
		                 
		        case 4 : System.exit(0);
		                 break;
		        
		        default: System.out.println("Invalid Choice..!");
		    }
		}
	}
}
